tests/sentry/ingest/test_outcome_consumer.py

```diff
@@ -348,17 +308,16 @@ def event_dropped_receiver(**kwargs):
     event_filtered.connect(event_filtered_receiver)
     event_dropped.connect(event_dropped_receiver)
 
+    consumer = get_outcomes_consumer(
+        max_batch_size=2, max_batch_time=100, group_id=group_id, auto_offset_reset="earliest"
+    )
+
     # run the outcome consumer
     with task_runner():
-        run_outcomes_consumer(
-            commit_batch_size=2,
-            consumer_group=consumer_group,
-            max_fetch_time_seconds=0.1,
-            initial_offset_reset="earliest",
-            is_shutdown_requested=_shutdown_requested(
-                max_secs=10, num_outcomes=1, signal_sink=event_filtered_sink
-            ),
-        )
+        i = 0
+        while len(event_dropped_sink) < 2 and i < 1000:
+            consumer._run_once()
```

Comment:
Since I have no `is_shutdown_requested` in `BatchingKafkaConsumer` I run my own while-loop in tests and call an internal function of `BatchingKafkaConsumer`.